plugins/org.eclipse.n4js/src/org/eclipse/n4js/postprocessing/ASTProcessor.xtend

```diff
@@ -436,6 +451,17 @@ public class ASTProcessor extends AbstractProcessor {
 				// process expression before varDeclOrBindings
 				obj.eContents.bringToFront(obj.expression)
 			}
+			ParameterizedCallExpression case MigrationUtils.isMigrateCall(obj): {
+				// link and type migration arguments first 
+				obj.eContents.bringToFront(obj.arguments)
+			}
+			Script case variantHelper.allowVersionedTypes(obj): {
+				// For variants that support versioned types and therefore migrations,
+				// process the migrations first.
```

Comment:
please add "process the migrations first, because information about signatures of all migrations will be required when processing migration calls".